### PR TITLE
Facility to View BTree as On-Disk Array

### PR DESCRIPTION
**Pull request by danking**
`OnDiskBTreeIndexToValue` can index the elements at the base of a BTree.

The BGen BTree is a BTree on the byte-offsets of variants in the BGen file. In a following PR, I will use this class to filter the BGen file to a subset of variants, specified by their index. This kind of filtering happens at the level of bytes, it permits me to avoid decoding/decompressing any variants I don't need.

Ideally, the index would also include the variant keys themselves. That would be a really nice extension of this work and would enable a more natural file-level filtering user experience.

Aside: `IndexBTree` could use some TLC. I'm sort of making the minimal changes to get Caitlin cooking. Last night, I slipped into a rewrite and it's just too big a task to get right before Friday because I don't have any documentation of precisely what our btree file format is.

### CODE REVIEW

**Comment by danking** [comment]
I need this reviewed today b/c I need to get the whole thing in master by Thursday so Caitlin can debug and run on Friday.

File: src/main/scala/is/hail/io/IndexBTree.scala [hunk 1]
```diff
@@ -62,6 +68,12 @@ object IndexBTree {
   ): Unit = hConf.writeDataFile(fileName) { w =>
     w.write(btreeBytes(arr, branchingFactor))
   }
+
+  def toString(
```
Comment (danking) [inline src/main/scala/is/hail/io/IndexBTree.scala]: all the above changes and this was just for some debugging help and better test failure messages

**Comment by jbloom22** [comment]
Dan, I won’t be able to review this until tomorrow afternoon (giving a talk).

**Comment by danking** [comment]
@rcownie reassigning since @jbloom22 is busy and need this preferable in this morning so more PRs can come after.

**Pull request merged**